src/coresyftools/coresyftools/gpt_tool.py

```diff
@@ -55,7 +61,8 @@ def run(self, bindings):
         bindings = bindings.copy()
         operator = self.operation.get('operation')
         if 'graph' in self.operation and self.operation['graph']:
-            graph_file = os.path.join(self.context_directory, self.DEFAULT_GPT_GRAPH_FILE_NAME)
+            graph_file_name = self._graph_file_path(self.operation)
+            graph_file = os.path.join(self.context_directory, graph_file_name)
```

Comment:
This step is also done inside method "_graph_file_path(operation)". Why to repeat here the "os.path.join" ?

Reply:
Good point. Indeed, this does not makes the implementation incorrect?
`graph_file` would not end up `<context_directory>/<context_directory>/<graph_file_name>`, where `context_directory` is the tool directory and `graph_file_name` the name of the gpt graph file?
I do not understand how the hell the test can pass.

Reply:
Ok, it works because in`os.path.join` _ If any component_ [argument] _is an absolute path, all previous components are thrown away, and joining continues._  So, the result of this line is to simply assign `graph_file_name` to `graph_file`.

Reply:
Fixed.